### PR TITLE
null  ContainerImage names bugfix

### PR DESCRIPTION
**Pull request by Avi-Robusta**
ContainerImages names can be None due to a containerd bug
we had to patch Hikaru - ContainerImage, and kubernetes python cli ContainerImage.
```
2022-11-10 12:15:57.973 ERROR    Failed to build execution event for update-Node-v1
2022-11-10 12:16:13.720 ERROR    Failed to run periodic nodes discovery
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/robusta/core/discovery/discovery.py", line 104, in discovery_process
    current_nodes: V1NodeList = client.CoreV1Api().list_node()
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api/core_v1_api.py", line 16414, in list_node
    return self.list_node_with_http_info(**kwargs)  # noqa: E501
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api/core_v1_api.py", line 16517, in list_node_with_http_info
    return self.api_client.call_api(
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 348, in call_api
    return self.__call_api(resource_path, method,
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 192, in __call_api
    return_data = self.deserialize(response_data, response_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 264, in deserialize
    return self.__deserialize(data, response_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 639, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 280, in __deserialize
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 280, in <listcomp>
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 639, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 639, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 280, in __deserialize
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 280, in <listcomp>
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 641, in __deserialize_model
    instance = klass(**kwargs)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/models/v1_container_image.py", line 55, in __init__
    self.names = names
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/models/v1_container_image.py", line 80, in names
    raise ValueError("Invalid value for `names`, must not be `None`")  # noqa: E501

```

### CODE REVIEW

File: src/robusta/patch/patch.py [hunk 1]
```diff
@@ -29,6 +30,17 @@ def create_monkey_patches():
...
     logging.info("Creating yaml monkey patch")
     YAML.official_plug_ins = official_plug_ins
+    # The patched method is due to a bug in containerd that allows for containerImages to have no names
+    # which causes the kubernetes python api to throw an exception
+    logging.info("Creating kubernetes ContainerImage monkey patch")
+    V1ContainerImage.names = V1ContainerImage.names.setter(names)
+
+
+def names(self, names):
+    if names:
```
Comment (arikalon1) [inline src/robusta/patch/patch.py:40]: This doesn't compile on my machine.

**Review by arikalon1 — APPROVED** [review]
Nice work.
Left one comment
Please fix and make sure it works before merging

File: src/robusta/patch/patch.py (hunk 1 above)
Comment (Avi-Robusta) [reply src/robusta/patch/patch.py:40]: there was a tab that got erased in the line below by acident

**Comment by Avi-Robusta** [comment]
testing on normal cluster to verify okay

**Pull request merged**